### PR TITLE
chore: Minor refactor

### PR DESCRIPTION
**Pull request by jamesgeorge007**
refactor code to a more concise version.

### CODE REVIEW

**Comment by jamesgeorge007** [comment]
cc @bcoe

**Comment by bcoe** [comment]
@jamesgeorge007 looks like a reasonable refactor, but tests are failing because `argv` does not exist in the context of the new helper; I might also be tempted to call the helper something closer to what it does.

As an example, `coerceValueType`, or something like that.

**Comment by jamesgeorge007** [comment]
@bcoe CI is still failing :thinking:

**Comment by jamesgeorge007** [comment]
@mleguen still the build doesn't turn green :thinking:

**Comment by bcoe** [comment]
@jamesgeorge007 if you clone this repository, and run:

```
npm i
npm t
```

Inside the folder, I think you should be able to see these exceptions happening on your local machine (without forcing CI/CD to run).

**Comment by bcoe** [comment]
👋 @jamesgeorge007 is this still something you'd like to see over the finish line?

**Comment by jamesgeorge007** [comment]
yeah

File: lib/validation.js [hunk 1]
```diff
@@ -209,43 +209,36 @@ module.exports = function validation (yargs, usage, y18n) {
...
     const implyFail = []
 
     Object.keys(implied).forEach((key) => {
       const origKey = key
       ;(implied[key] || []).forEach((value) => {
-        let num
         let key = origKey
         const origValue = value
+        // eslint-disable-next-line
+        [key, value] = [key, value].map((item) => coerceValueType(argv, item))
```
Comment (jamesgeorge007) [inline lib/validation.js]: @bcoe It seems we aren't getting the intended values while destructuring `key` and `value`.

Comment (mleguen) [reply lib/validation.js]: The issue disappears if you do not use destructuring, but neither do I understand why, nor am I able to reproduce such a destructuring issue in a simpler code.

```suggestion
        key = coerceValueType(argv, key)
        value = coerceValueType(argv, value)
```

But in the end, IMHO, having 2 lines is more readable than using a map here...

File: lib/validation.js [hunk 2]
```diff
@@ -209,43 +209,36 @@ module.exports = function validation (yargs, usage, y18n) {
     return implied
   }
 
+  function coerceValueType (argv, val) {
```
Comment (mleguen) [inline lib/validation.js]: I suggest renaming this function, as it is not coercing the value type, but checking its presence.

```suggestion
  function keyExists (argv, key) {
```

File: lib/validation.js (hunk 1 above)
Comment (mleguen) [reply lib/validation.js]: And this way you can also drop the "please don't lint my hack" hint above ;-)

**Comment by bcoe** [comment]
@mleguen feel free to land this when you're happy?

File: lib/validation.js [hunk 3]
```diff
@@ -209,43 +209,37 @@ module.exports = function validation (yargs, usage, y18n) {
...
   self.implications = function implications (argv) {
     const implyFail = []
 
     Object.keys(implied).forEach((key) => {
       const origKey = key
       ;(implied[key] || []).forEach((value) => {
-        let num
         let key = origKey
         const origValue = value
+        // eslint-disable-next-line
```
Comment (mleguen) [inline lib/validation.js]: This hint should no longer be useful

**Comment by mleguen** [comment]
Thanks @jamesgeorge007 

Could you please remove the hint which is no longer useful before I merge this?

**Comment by jamesgeorge007** [comment]
@mleguen done :+1:

**Comment by mleguen** [comment]
Thanks @jamesgeorge007

**Pull request merged**